services/tracing.py

```diff
@@ -93,24 +101,19 @@ def _set_attributes(span, attributes_dict):
         @wraps(func)
         def wrap_with_span(*args, **kwargs):
             name = span_name or func.__qualname__
+            attributes_dict = attributes or {}
+            if func_attributes is not None:
+                func_signature = signature(func)
+                bound_args = func_signature.bind(*args, **kwargs).arguments
+                for k, v in func_attributes.items():
+                    func_arg = _get_argument_from_function_signature(
+                        func_signature, bound_args, kwargs, v
+                    )
+                    attributes_dict[k] = str(func_arg)
+
             with tracer.start_as_current_span(
                 name, record_exception=record_exception
```

Comment:
You can pass attributes at span creation, so we could maybe do

```
with tracer.start_as_current_span(name, record_exception=record_exception, attributes=attributes_dict):```

This would remove the need for _set_attributes alltogether?